replayserver/bookkeeping/bookkeeper.py

```diff
@@ -32,5 +36,35 @@ def build(cls, database, config):
             await self._saver.save_replay(game_id, stream)
             logger.debug(f"Saved replay {game_id}")
             metrics.saved_replays.inc()
+            logger.debug(f"Parsing replay {game_id}")
+            ticks = await self.get_ticks(stream)
+            await self._queries.update_game_stats(game_id, ticks)
         except BookkeepingError as e:
             logger.warning(f"Failed to save replay for game {game_id}: {e}")
+
+    async def get_ticks(self, stream):
+        """Parse the replay and extract the total number of ticks"""
+        loop = asyncio.get_event_loop()
+        return await loop.run_in_executor(
```

Comment:
Why do we need this?

Reply:
Maybe not. I'll run the profile tests

Reply:
If the intention was to run body parsing in a separate loop, then it won't work, since replay parser is pure Python code and does not release the GIL. If it was a C extension (like zlib mostly is), then it could do that.

Reply:
Yea, it appears to make no difference when running your profile tests